### PR TITLE
speedup RetryTrait Tests

### PR DESCRIPTION
**Pull request by verfriemelt-dot-org**
i think its a good idea not to wait on actual retry times during the tests

### CODE REVIEW

File: src/Api/RetryTrait.php [hunk 1]
```diff
@@ -46,11 +50,23 @@ private function retry(callable $callback, int $maxRetries = 3, bool $addJitter
...
-            $waitTime = 1 == $attempt ? 100 : 2 ** $attempt * 100; // exponential strategy
-            $waitTime = $addJitter ? random_int(0, $waitTime) : $waitTime; // adding randomness
-            // wait
-            usleep($waitTime * 1000);
+        if ($attempt <= 0) {
+            return;
+        }
+
+        // exponential strategy
+        $waitTime = (1 + 2 ** $attempt) * 100;
```
Comment (jschaedl) [inline src/Api/RetryTrait.php:58]: Smart :-)

File: src/Api/RetryTrait.php [hunk 2]
```diff
@@ -46,11 +50,23 @@ private function retry(callable $callback, int $maxRetries = 3, bool $addJitter
...
+
+        if ($addJitter) {
+            // jitter by +/- 25%
+            $waitTime *= random_int(75, 125) / 100;
         }
+
+        usleep((int) $waitTime * static::backupTimeBaseFactor());
+    }
+
+    private function backupTimeBaseFactor(): int
```
Comment (jschaedl) [inline src/Api/RetryTrait.php]: nice idea

Comment (verfriemelt-dot-org) [reply src/Api/RetryTrait.php]: i am not sure about the method name, couldn't come up with something more suiteable :thinking:

Comment (jschaedl) [reply src/Api/RetryTrait.php]: Maybe just `getBaseFactor()`?

Comment (verfriemelt-dot-org) [reply src/Api/RetryTrait.php]: i have settled for `getWaitTimeFactor()` to make it what it will return :)

**Comment by jschaedl** [comment]
Thank you @verfriemelt-dot-org !

**Pull request merged**